### DISCUSSION TITLE
How to use extras for psycopg2 and pyscopg2-binary

### DISCUSSION THREAD

**Discussion by 1oglop1** [python-poetry/poetry#4432] [Q&A]
Hi, we are using `psycopg2-binary` for local development installed via `requirements-dev.txt` and `psycopg2` installed on the server after deployment.

From what I understand, I should be able to achieve this with poetry as well however I'm not sure how to configure and run it.

Can some please share an example?

I found the test for this so I reckon it should be possible.

https://github.com/python-poetry/poetry/blob/b753aaf4c3c08ef0e54941a6616fe318fdf4f6e4/tests/repositories/test_pypi_repository.py#L154

I'd like to be able to install `psycopg2-binary` in my dev env and `psycopg2` for deployment.

Thank you!

**Comment by britodfbr** [comment]
Poetry add -D psycopg2-binary
Poetry add psycopg2

**Reply by 1oglop1** [reply]
I do not think that this does the trick.
The issue will likely be resolved by https://github.com/python-poetry/poetry/issues/1644
